# Python Syntax

Python bergantung pada identitation.
* Indentation dalam Python artinya jarak masuk/spasi di awal baris kode untuk menunjukkan bahwa suatu baris merupakan bagian dari blok kode tertentu.

In [1]:
age = 25
if age < 18:
    print("You are a minor.")

```
if age < 18:
    print("You are a minor.")
    ↑
    indentation
```

# Comment

In [ ]:
# This commetn
age = 25

# Multiple comments
"""
This is a multi-line comment
this is a multi-line comment
"""

# Variables

In [2]:
name = "Andi"
age = 25
salary = 7500000
is_employee = True

In [3]:
type(name)  # <class 'str'>
type(age)  # <class 'int'>
type(salary)  # <class 'int'>
type(is_employee)  # <class 'bool'>


bool

# Primitive Data Types

## Integer

In [4]:
age = 35
quantity = 1000

print(type(age))  # <class 'int'>
print(type(quantity))  # <class 'int'>

<class 'int'>
<class 'int'>


## Float

In [5]:
price = 102.222
temperature = 36.6

print(type(price))  # <class 'float'>
print(type(temperature))  # <class 'float'>

<class 'float'>
<class 'float'>


In [8]:
print(10 / 3)
print(10 // 3)

3.3333333333333335
3


* / = true division
* // = floor division

# Floating Point Precission

In [9]:
print(0.1 + 0.2)

0.30000000000000004


hasil bukan 0.3 tetapi 0.30000000000000004 > karena angka desimal tertentu dapat dipresentasikan secara tepat dlaam format floating-point biner

In [10]:
print(0.1 + 0.2 == 0.3)

False


In [ ]:
# Untuk kebutuhan tertentu

import math

math.isclose(0.1 + 0.2, 0.3) 

# == tidak selalu aman untuk membandingkan float, gunakan math.isclose() untuk membandingkan float

True

# String

In [13]:
nama1 = "Andi"
nama2 = 'Budi'

print(nama1)
print(nama2)

Andi
Budi


## String Indexing

In [ ]:
name = "Python"

name[0]

'P'

In [15]:
name[-1]

'n'

# String Slicing

In [16]:
kata = "Python"

kata[0:3]

'Pyt'

pola string[start:stop]

* stop tidak termasu

# String method

In [ ]:
text = "            Python is a programming language"


Python is a programming language


## strip()

menghilangkan spasi di awal kata

In [18]:
print(text.strip()) 

Python is a programming language


## lower()

mengubah semua huruf menjadi kecil

In [19]:
text.lower()

'            python is a programming language'

## upper()

* mengubah huruf menjadi kapital

In [20]:
text.upper()

'            PYTHON IS A PROGRAMMING LANGUAGE'

## replace()

* mengganti kata

In [21]:
text.replace("Python", "Java")

'            Java is a programming language'

## split()

* memisahkan data

In [ ]:
"Python,Pandas,NumPy".split(",")

## joint()

In [23]:
languages = ['python', 'java', 'c++', 'javascript']

", ".join(languages)

'python, java, c++, javascript'

# F-string

* Formatted string literal: cara di python memasukan nilai variabel  atau hasil operasi langsung ke dalam sebuah string

In [24]:
name = "John"
age = 30

print(f"My name is {name} and I am {age} years old.")

My name is John and I am 30 years old.


```
f"Halo, {name}!"
          ↑
      ambil nilai name
```

In [25]:
price = 10000
quantity = 3

print(f"Total: {price * quantity}")

Total: 30000


In [26]:
name = "budi"

print(f"Nama: {name.upper()}")

Nama: BUDI


In [27]:
score = 0.85678

print(f"Score: {score:.2f}")

Score: 0.86



:.2f berarti:

* f → format sebagai floating-point
* .2 → tampilkan 2 angka di belakang koma

In [28]:
mae = 19.322
rmse = 25.781
r2 = 0.89

print(f"MAE  : {mae:.2f}")
print(f"RMSE : {rmse:.2f}")
print(f"R²   : {r2:.2%}")

MAE  : 19.32
RMSE : 25.78
R²   : 89.00%
